src/main/java/teammates/common/datatransfer/questions/FeedbackMsqResponseDetails.java

```diff
@@ -85,7 +85,59 @@ public String getAnswerCsv(FeedbackQuestionDetails questionDetails) {
 
     @Override
     public List<String> validateResponseDetails(FeedbackQuestionAttributes correspondingQuestion) {
-        return new ArrayList<>();
+        List<String> errors = new ArrayList<>();
+        List<String> msqChoices = ((FeedbackMsqQuestionDetails) correspondingQuestion
+                .getQuestionDetails()).getMsqChoices();
+        int maxSelectableChoices = ((FeedbackMsqQuestionDetails) correspondingQuestion
+                .getQuestionDetails()).getMaxSelectableChoices();
+        int minSelectableChoices = ((FeedbackMsqQuestionDetails) correspondingQuestion
+                .getQuestionDetails()).getMinSelectableChoices();
+        boolean isOtherEnabled = ((FeedbackMsqQuestionDetails) correspondingQuestion
+                .getQuestionDetails()).getOtherEnabled();
```

Comment:
You should consider to extract `((FeedbackMsqQuestionDetails) correspondingQuestion.getQuestionDetails())` to a variable.